openedx/features/pakx/lms/pakx_admin_app/views.py

```diff
@@ -419,3 +421,30 @@ def get(self, *args, **kwargs):  # pylint: disable=unused-argument
             user_info['role'] = TRAINING_MANAGER if user_groups.name == GROUP_TRAINING_MANAGERS else ORG_ADMIN
 
         return Response(status=status.HTTP_200_OK, data=user_info)
+
+
+class CourseListAPI(generics.ListAPIView):
+    authentication_classes = [SessionAuthentication]
+    permission_classes = [CanAccessPakXAdminPanel]
+    pagination_class = PakxAdminAppPagination
+    serializer_class = CoursesSerializer
+
+    PakxAdminAppPagination.page_size = 3
+
+    instructors = {}
+
+    def get_serializer_context(self):
+        context = super(CourseListAPI, self).get_serializer_context()
+        context.update({"instructors": self.instructors})
+        return context
+
+    def get_queryset(self):
+        queryset = CourseOverview.objects.all()
+        course_access_role_qs = CourseAccessRole.objects.filter(course_id__in=queryset.values_list('id'))
+        for course_access_role in course_access_role_qs:
+            if self.instructors.get(course_access_role.course_id):
+                self.instructors[course_access_role.course_id][course_access_role.user.profile.name] = None
+                continue
+            self.instructors[course_access_role.course_id] = {course_access_role.user.profile.name: None}
```

Comment:
try this, the list is no suitable instead of dict since you are not using values 
```
        course_instructors = self.instructors.get(course_access_role.course_id, [])
        course_instructors.append(course_access_role.user.profile.name)
        self.instructors[course_access_role.course_id] = course_instructors
```